XamlControlsGallery/ControlPages/ProgressBarPage.xaml

```diff
@@ -47,9 +47,15 @@
                 <muxc:ProgressBar Width="130" x:Name="ProgressBar2"/>
                 <TextBlock x:Name="Control2Output" Style="{ThemeResource OutputTextBlockStyle}" Width="60" TextAlignment="Center" />
                 <TextBlock x:Name="ProgressLabel"  Text="Progress" VerticalAlignment="Center" Margin="0,0,10,0"/>
-                <muxc:NumberBox x:Name="ProgressValue" AutomationProperties.LabeledBy="{Binding ElementName=ProgressLabel}"
-                    Minimum="0" Maximum="100" Value="0"
-                    SpinButtonPlacementMode="Inline" ValueChanged="ProgressValue_ValueChanged"/>
+                <muxc:NumberBox
+                    x:Name="ProgressValue"
+                    AutomationProperties.LabeledBy="{Binding ElementName=ProgressLabel}"
+                    AutomationProperties.Name="NumberBox controlling ProgressBar2 value"
```

Comment:
Shouldn't setting the LabeledBy property be enough here? Could this actually be an issue with the NumberBox control?

Reply:
LabeledBy applies the automationProperties.Name property to NumberBox but not the textbox, but mysteriously, the .Name property applies the name to both. This behaviour is duplicated with ComboBox. This is a temporary fix while I investigate the discrepancies between the two properties.

Reply:
I see, very strange that it works for one of the properties but not for the other. Could we template bind to the labeled by property maybe to mitigate this?

Reply:
One option is to templatebind InputBox to the header name like ComboBox, but that still doesn't solve the issue if there is no header available.